### PR TITLE
Add documentation for the diff.

### PR DESCRIPTION
**Pull request by polina-c**
See go/devtools-snapshots-help-wording for text

Screen Shot 2022-11-01 at 10 55 04 AM

### CODE REVIEW

**Review by bkonyi — APPROVED** [review]
Generally LGTM, but we might want to get some feedback on the documentation wording.

File: packages/devtools_app/lib/src/screens/memory/panes/diff/diff_pane.dart [hunk 1]
```diff
@@ -71,3 +85,21 @@ class _SnapshotItemContent extends StatelessWidget {
...
+Take a heap snapshot to view the current memory allocation details.
+
+1. Click on the ● button in the 'Snapshots' panel to the left (or click [ ● Take Snapshot ] below these instructions)
+2. Use the filter button above the snapshot table to refine the results
+3. Select a class from the snapshot table to view its retaining paths in another table
+4. Select a path from the 'Shortest retaining paths…' table to view the path details
+
+View the diff between snapshots to detect or debug allocation issues for a feature.
+
+1. Take a snapshot (1) at the starting point of the feature in your application
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/panes/diff/diff_pane.dart:97]: @MaryaBelanger , involving you, as original editor, too.

[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/panes/diff/diff_pane.dart:97]: Be, can you give details on what exactly does not look good enough for you?

[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/panes/diff/diff_pane.dart:97]: Started discussion on this on Chat. Will follow up with edits in separate PR.

[the remark this answers is not in the document]
Comment (sfshaza2) [reply packages/devtools_app/lib/src/screens/memory/panes/diff/diff_pane.dart:97]: Great! Thanks, @polina-c!

**Pull request merged**